device/blackhole/blackhole_coordinate_manager.cpp

```diff
@@ -0,0 +1,14 @@
+#include "blackhole_coordinate_manager.h"
+
+std::set<std::size_t> BlackholeCoordinateManager::get_x_coordinates_to_harvest(std::size_t harvesting_mask) {
```

Comment:
Can we use [`std::bit_set`](https://en.cppreference.com/w/cpp/utility/bitset) for this?

Reply:
What problem does that solve?

Reply:
Went with `__builtin_popcount` for `std::size_t`